src/App/ViewModels/Settings/ConnectionSettingsViewModel.cs

```diff
@@ -105,6 +105,12 @@ private void OnSettingsPropertyChanged(object sender, System.ComponentModel.Prop
                 SDK.AllowLocalAdaptersWhenConnected = Properties.Settings.Default.AllowLanInterfaces;
                 Properties.Settings.Default.Save();
             }
+
+            if (e.PropertyName == nameof(Properties.Settings.Default.IsThreatProtectionEnabled))
+            {
+                SDK.DnsFilterMode = IsThreatProtectionEnabled ? VpnSDK.Enums.DnsFilteringMode.WithWLVPNDns : VpnSDK.Enums.DnsFilteringMode.Disabled; 
+                Properties.Settings.Default.Save();
```

Comment:
"Properties.Settings.Default.Save();" can be avoided as there is no change in the settings properties